lib/posix/options/Kconfig.deprecated

```diff
@@ -6,6 +6,27 @@
 
 menu "Deprecated POSIX options"
 
+config MAX_TIMER_COUNT
+	int "Maximum number of timer_t [DEPRECATED]"
+	default 0
+	help
```

Comment:
Could you add `select DEPRECATED` to all deprecated kconfig options where it is missing.